include/openthread/srp_server.h

```diff
@@ -399,6 +445,17 @@ const char *otSrpServerHostGetFullName(const otSrpServerHost *aHost);
  */
 const otIp6Address *otSrpServerHostGetAddresses(const otSrpServerHost *aHost, uint8_t *aAddressesNum);
 
+/**
+ * This function returns the LEASE and KEY-LEASE information of a given host.
+ *
+ * @param[in]  aHost  A pointer to the SRP server host.
+ *
+ * @returns  An otSrpServerLeaseInfo of the host. It contains the LEASE time, KEY-LEASE time, remaining LEASE time and
+ * the remaining KEY-LEASE time.
+ *
+ */
+otSrpServerLeaseInfo otSrpServerHostGetLeaseInfo(const otSrpServerHost *aHost);
```

Comment:
This API is returning an `struct` (same with the `otSrpServerServiceGetLeaseInfo()`). While this works ok, it is not a good/efficient model in C (or C++) sicne it cause copying of the `struct`. 

I suggest changing the signature for caller to pass in `otSrpServerLeaseInfo *` to be populated.

Reply:
Thanks for the suggestion!